sklearn/datasets/species_distributions.py

```diff
@@ -230,7 +230,13 @@ def fetch_species_distributions(data_home=None,
 
         print('Downloading species data from %s to %s' % (SAMPLES_URL,
                                                           data_home))
-        X = np.load(BytesIO(urlopen(SAMPLES_URL).read()))
+        expected_samples_checksum = ("abb07ad284ac50d9e6d20f1c4211e0fd3c098f7f"
+                                     "85955e89d321ee8efe37ac28")
+        samples_path = join(data_home, "samples.zip")
+        _fetch_url(SAMPLES_URL, samples_path,
+                   expected_samples_checksum)
+        X = np.load(samples_path)
```

Comment:
Am I right to think this is a rather obscure usage of `np.load`? I suppose this fully unpacks the zip file into memory, making it safe to remove the path (even on Windows)?

Perhaps this deserves a comment, as the docstring for numpy.load isn't very helpful.